### PR TITLE
Fix problems with Credentials error in uploads

### PR DESCRIPTION
**Pull request by masensio**

### CODE REVIEW

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 1]
```diff
@@ -1025,6 +1027,10 @@ public void onRemoteOperationFinish(RemoteOperation operation, RemoteOperationRe
             onGetUserNameFinish(result);
         }
 
+        if (result.isSuccess() && mAction == ACTION_UPDATE_EXPIRED_TOKEN) {
+            setResult(FileActivity.RESULT_OK);
```
Comment (davivel) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Could we use ```Activity.RESULT_OK```?  That way we don't need to add a dependency from ```AuthenticatorActivity``` to ```FileActivity```

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 2]
```diff
@@ -1419,6 +1425,9 @@ public void onAuthenticatorTaskCallback(RemoteOperationResult result) {
             }
 
             if (success) {
+                if (mAction == ACTION_UPDATE_EXPIRED_TOKEN) {
+                    setResult(FileActivity.RESULT_OK);
```
Comment (davivel) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Same here

File: src/com/owncloud/android/files/FileOperationsHelper.java [hunk 3]
```diff
@@ -541,6 +543,17 @@ public void retryUpload(OCUpload upload) {
         }
     }
 
+    public void retryUploadsForAccount(Account account) {
```
Comment (davivel) [inline src/com/owncloud/android/files/FileOperationsHelper.java]: Could we move this method to ```FileUploader```? As a static method ```retryUploadsForAccount(Account account, Context context)```, similar to the new static methods to ease the creation of upload requests. Thanks.

File: src/com/owncloud/android/ui/activity/FileActivity.java [hunk 4]
```diff
@@ -833,7 +834,7 @@ protected void requestCredentialsUpdate(Context context) {
                     AuthenticatorActivity.EXTRA_ACTION,
                     AuthenticatorActivity.ACTION_UPDATE_EXPIRED_TOKEN);
             updateAccountCredentials.addFlags(Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
-            startActivity(updateAccountCredentials);
+            startActivityForResult(updateAccountCredentials, UPDATE_CREDENTIALS_REQUEST_CODE);
```
Comment (davivel) [inline src/com/owncloud/android/ui/activity/FileActivity.java]: This change is not needed anymore, right?

File: src/com/owncloud/android/ui/activity/FileActivity.java [hunk 5]
```diff
@@ -114,6 +113,8 @@
 
     protected static final long DELAY_TO_REQUEST_OPERATIONS_LATER = 200;
 
+    public static final int UPDATE_CREDENTIALS_REQUEST_CODE = 100;
```
Comment (davivel) [inline src/com/owncloud/android/ui/activity/FileActivity.java]: I'd move this constant to ```UploadListActivity```, that's the Activity really needing it.

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java (hunk 1 above)
Comment (masensio) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Of course, we can use Activity.RESULT_OK but this change involves adding the import android.app.Activity too.

File: src/com/owncloud/android/files/FileOperationsHelper.java (hunk 3 above)
Comment (masensio) [inline src/com/owncloud/android/files/FileOperationsHelper.java]: Done

File: src/com/owncloud/android/ui/activity/FileActivity.java (hunk 4 above)
Comment (masensio) [inline src/com/owncloud/android/ui/activity/FileActivity.java]: Right.

**Comment by masensio** [comment]
Close this pull request and create another new with relevant commits https://github.com/owncloud/android/commit/4d57de703245494c1ee228df927e08c2390f9a01 and https://github.com/owncloud/android/commit/ec709eca98a506fdea9520dd54adebcba0d71583

**Pull request closed without merging**